GameOfRunes/GameOfRunes/GameEngine/Entities/AttractionEntity.swift

```diff
@@ -13,7 +13,7 @@ class AttractionEntity: Entity {
         .attractionEntity
     }
     
-    init(node: SKSpriteNode, layerType: SpriteLayerType, team: Team) {
+    init(node: SKSpriteNode, layerType: SpriteLayerType, team: Team, parent: Entity) {
```

Comment:
Hmm, perhaps make `parent` optional? Currently, this implementation seems to enforce that `AttractionEntity` cannot exist by itself. E.g. it must be tagged to a parent entity. But we should allow it to exist by itself right?

Reply:
Yup that's true, I thought that it would make sense for it to be tagged to a parent since I was thinking to remove the `spritenode`

Reply:
I think another prob also lies with the spritenode being used to hold `size`, `position` and `view` info. By right, should have 1 component for each. But its okay uh hahahaha.